src/gt4py/next/ffront/foast_to_gtir.py

```diff
@@ -0,0 +1,384 @@
+                return im.op_as_fieldop("if_")(cond, true_, false_)
+
+            return create_if_impl
+
+        result = lowering_utils.process_elements(
+            create_if(im.ref(cond_symref_name)),
+            (self.visit(node.args[1]), self.visit(node.args[2])),
+            node.type,
+        )
+
+        return im.let(cond_symref_name, cond_)(result)
+
+    _visit_concat_where = _visit_where  # TODO(havogt): upgrade concat_where
+
+    def _visit_broadcast(self, node: foast.Call, **kwargs: Any) -> itir.FunCall:
+        return self.visit(node.args[0], **kwargs)
+
+    def _visit_math_built_in(self, node: foast.Call, **kwargs: Any) -> itir.FunCall:
+        return self._map(self.visit(node.func, **kwargs), *node.args)
+
+    def _make_reduction_expr(
+        self, node: foast.Call, op: str | itir.SymRef, init_expr: itir.Expr, **kwargs: Any
+    ) -> itir.Expr:
+        # TODO(havogt): deal with nested reductions of the form neighbor_sum(neighbor_sum(field(off1)(off2)))
+        it = self.visit(node.args[0], **kwargs)
+        assert isinstance(node.kwargs["axis"].type, ts.DimensionType)
+        val = im.call(im.call("reduce")(op, init_expr))
+        return im.op_as_fieldop(val)(it)
+
+    def _visit_neighbor_sum(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
+        dtype = type_info.extract_dtype(node.type)
+        return self._make_reduction_expr(node, "plus", self._make_literal("0", dtype), **kwargs)
+
+    def _visit_max_over(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
+        dtype = type_info.extract_dtype(node.type)
+        min_value, _ = type_info.arithmetic_bounds(dtype)
+        init_expr = self._make_literal(str(min_value), dtype)
+        return self._make_reduction_expr(node, "maximum", init_expr, **kwargs)
+
+    def _visit_min_over(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
+        dtype = type_info.extract_dtype(node.type)
+        _, max_value = type_info.arithmetic_bounds(dtype)
+        init_expr = self._make_literal(str(max_value), dtype)
+        return self._make_reduction_expr(node, "minimum", init_expr, **kwargs)
+
+    def _visit_type_constr(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
+        if isinstance(node.args[0], foast.Constant):
+            node_kind = self.visit(node.type).kind.name.lower()
+            target_type = fbuiltins.BUILTINS[node_kind]
+            source_type = {**fbuiltins.BUILTINS, "string": str}[node.args[0].type.__str__().lower()]
+            if target_type is bool and source_type is not bool:
+                return im.literal(str(bool(source_type(node.args[0].value))), "bool")
+            return im.literal(str(node.args[0].value), node_kind)
+        raise FieldOperatorLoweringError(
+            f"Encountered a type cast, which is not supported: {node}."
+        )
+
+    def _make_literal(self, val: Any, type_: ts.TypeSpec) -> itir.Expr:
+        if isinstance(type_, ts.ScalarType):
```

Comment:
The tuple case that used to be here is most likely for `tuple`s originating from `FrozenNamespace`s.

Reply:
added a test and the code back